src/Hyperion/Run.hs

```diff
@@ -81,6 +89,20 @@ runBenchmark samplef bk0 =
       put xs
       return x
 
+-- | Run a benchmark on varying sample sizes
+runBenchmarkVaryingSizes :: [(Batch () -> IO Sample)] -> Benchmark -> IO (HashMap Text [Sample])
+runBenchmarkVaryingSizes samplef bk0 = formatSamples $ sequence (map (\samplef' -> runBenchmark samplef' bk0) samplef)
+
+-- | Set of functions to massage the data into the appropriate type
+formatSamples :: IO([HashMap Text Sample]) -> IO (HashMap Text [Sample])
+formatSamples samples = concatenateSamples $ expandSamples samples
+
+expandSamples :: IO ([HashMap Text Sample]) -> IO ([HashMap Text [Sample]])
+expandSamples samples = fmap (map (fmap (\x -> [x]))) samples
```

Comment:
In general it's very uncommon lift functions to the monad like that. Try to keep your code as pure as possible.

You can write (applying η-reduce):
```
expandSamples :: [HashMap Text Sample]) -> [HashMap Text [Sample]])
expandSamples = map (fmap (\x -> [x])) 
```

and then lift it to IO when needed as `concatenateSamples $ fmap expandSamples samples`